Sources/ParseSwift/Authentication/3rd Party/ParseFacebook.swift

```diff
@@ -0,0 +1,251 @@
+//
+//  ParseFacebook.swift
+//  ParseSwift
+//
+//  Created by Abdulaziz Alhomaidhi on 3/18/21.
+//  Copyright © 2021 Parse Community. All rights reserved.
+
+import Foundation
+#if canImport(Combine)
+import Combine
+#endif
+
+// swiftlint:disable line_length
+
+/**
+ Provides utility functions for working with Facebook User Authentication and `ParseUser`'s.
+ Be sure your Parse Server is configured for [sign in with Facebook](https://docs.parseplatform.org/parse-server/guide/#configuring-parse-server-for-sign-in-with-facebook).
+ For information on acquiring Facebook sign-in credentials to use with `ParseFacebook`, refer to [Facebook's Documentation](https://developers.facebook.com/docs/facebook-login/limited-login.
+ */
+public struct ParseFacebook<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Facebook authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case token
+
+        /// Properly makes an authData dictionary with the required keys.
+        /// - parameter facebookId: Required id for the user.
+        /// - parameter authToken: Required identity token for the user.
+        /// - returns: Required authData dictionary.
+        func makeDictionary(facebookId: String,
+                            authToken: String) -> [String: String]? {
+
+            return [AuthenticationKeys.id.rawValue: facebookId,
+             AuthenticationKeys.token.rawValue: authToken]
+        }
+
+        /// Verifies all mandatory keys are in authData.
+        /// - parameter authData: Dictionary containing key/values.
+        /// - returns: `true` if all the mandatory keys are present, `false` otherwise.
+        func verifyMandatoryKeys(authData: [String: String]?) -> Bool {
+            guard let authData = authData,
+                  authData[AuthenticationKeys.id.rawValue] != nil,
+                  authData[AuthenticationKeys.token.rawValue] != nil else {
+                return false
+            }
+            return true
+        }
+    }
+    public static var __type: String { // swiftlint:disable:this identifier_name
+        "facebook"
+    }
+    public init() { }
+}
+
+// MARK: Login
+public extension ParseFacebook {
+    /**
+     Login a `ParseUser` *asynchronously* using Facebook authentication.
+     - parameter withFacebookId: The `Facebook userId` from `FacebookSDK`.
+     - parameter authToken: The `authToken` from `FacebookSDK`.
+     - parameter options: A set of header options sent to the server. Defaults to an empty set.
+     - parameter callbackQueue: The queue to return to after completion. Default value of .main.
+     - parameter completion: The block to execute.
+     */
+
+    func login(withFacebookId: String,
+               authToken: String,
```

Comment:
`withFacebookId` should be `facebookId`. 

After you change `authToken`, you should make a copy of this function, but it should take `accessToken` instead of `authenticationToken`. The documentation comments above should describe is the method is for limited login or not. This should be done for the sync, async, and publisher methods